core/templates/pages/exploration-player-page/services/stats-reporting.service.ts

```diff
@@ -109,6 +109,11 @@ export class StatsReportingService {
     if (StatsReportingService.explorationIsComplete) {
       return;
     }
+    let stateStatsMapping = (
+      StatsReportingService.aggregatedStats.state_stats_mapping);
+    if (stateStatsMapping.hasOwnProperty(undefined)) {
+      throw new Error('Aggregated stats contains an undefined state name');
```

Comment:
You don't want this to result in a user-facing error, right? (It shouldn't block the user flow.) If so, then $log.error() and then returning is enough, I think.

Reply:
I've added a backend log for this instead.